xt/testapp/t/004_selenium.t

```diff
@@ -0,0 +1,268 @@
+                # submit the username first
+                if ($selenium_config->{"separate_passwd_page"}) {
+                    # Find the login button and click it
+                    my $login = $driver->find_element(
+                                                $selenium_config->{"login_btn"},
+                                                $selenium_config->{"login_btn_by"});
+                    next if ( ! $login);
+
+                    $login->click('left');
+                    $driver->pause(500);
+                }
+
+                # Find the password field and enter the password
+                my $password = $driver->find_element(
+                                                $selenium_config->{"password_fn"},
+                                                defined $selenium_config->{"password_by"} ?
+                                                $selenium_config->{"password_by"} : "id"
+                                                );
+                next if ( ! $password );
+
+                $password->send_keys($selenium_config->{'password'});
+
+                # Find the login button and click it
+                my $login = $driver->find_element(
+                                                $selenium_config->{"login_btn"},
+                                                $selenium_config->{"login_btn_by"});
+                next if ( ! $login);
+
+                $login->click('left');
+                $driver->pause(500);
+
+                # Check for a post login IdP page and click the continue button
+                if ( defined $selenium_config->{"post_login_page_title"} &&
+                                $driver->get_title eq $selenium_config->{"post_login_page_title"} )
+                {
+                    print "Found post login page - continuing\n";
+                    my $proceed = $driver->find_element(
+                                                $selenium_config->{"post_login_page_btn"},
+                                                $selenium_config->{"post_login_page_btn_by"});
+                    next if ( ! $proceed );
+                    $proceed->click('left');
+                    $driver->pause(500);
+                }
+            } else {
+                # The IdP automatically logged in the user - it considered
+                # the last session as still active.
+                print "    Automatically logged in!!!\n"
+            }
+            print "    Login Complete\n";
+
+            # Check the value of the issuer if issuer_value is defined in the config
+            my $value = $driver->find_element_by_id("issuer");
+            if ( defined $selenium_config->{"issuer_value"} ) {
+                ok ($selenium_config->{"issuer_value"} eq $value->get_text(), "Login confirmed: $selenium_config->{'issuer_value'} found");
+            }
+
+            print "    Logout using: $logout_binding", "\n";
```

Comment:
As this is a testfile, I would opt to use `diag` or `note`.

Reply:
sounds good